api/src/main/java/org/openmrs/api/OrderService.java

```diff
@@ -724,6 +724,17 @@ public Order discontinueOrder(Order orderToDiscontinue, String reasonNonCoded, D
 	 */
 	@Authorized(PrivilegeConstants.GET_ORDER_TYPES)
 	public OrderType getOrderTypeByConcept(Concept concept);
+
+	/**
+	 * Get order types by java class name
+	 *
+	 * @param javaClassName the class name used to get the order types
+	 * @since 2.7.0
+	 * @return return the order types associated with given class name
+	 * <strong>Should</strong> find order types with the specified class name
+	 */
+	@Authorized(PrivilegeConstants.GET_ORDER_TYPES)
+	public List<OrderType> getOrderTypesByClassName(String javaClassName) throws APIException;
```

Comment:
These service and DAO methods seem like a good addition.  We will need to update the since annotation to reflect the right versions at this point.  However, I think we should also consider how these methods support inheritance (or don't currently).  For example, maybe I have an Order class in my module that extends TestOrder.  If I call this method with "org.openmrs.TestOrder", should that return just OrderTypes that have a class of "org.openmrs.TestOrder", or should it also return OrderTypes that have a class of something that extends "org.openmrs.TestOrder"?  Do we want to add that as a possible option via an argument?